src/Roots/Acorn/Bootloader.php

```diff
@@ -259,20 +266,22 @@ public function getApplication(): ApplicationContract
     {
         $this->app ??= new Application($this->basePath(), $this->usePaths());
 
-        $this->app->singleton(
-            \Illuminate\Contracts\Http\Kernel::class,
-            \Roots\Acorn\Kernel::class
-        );
-
-        $this->app->singleton(
-            \Illuminate\Contracts\Console\Kernel::class,
-            \Roots\Acorn\Console\Kernel::class
-        );
-
-        $this->app->singleton(
-            \Illuminate\Contracts\Debug\ExceptionHandler::class,
-            \Roots\Acorn\Exceptions\Handler::class
-        );
+        $httpKernel = Env::get('ACORN_ENABLE_EXPIRIMENTAL_ROUTER')
+            ? \Roots\Acorn\Http\Kernel::class
+            : \Roots\Acorn\Kernel::class;
+
+        collect([
+            \Illuminate\Contracts\Http\Kernel::class => $httpKernel,
+            \Illuminate\Contracts\Console\Kernel::class => \Roots\Acorn\Console\Kernel::class,
+            \Illuminate\Contracts\Debug\ExceptionHandler::class => \Roots\Acorn\Exceptions\Handler::class,
+        ])
+        ->map(fn($concrete, $abstract) => apply_filters(
+            "acorn/container/"
+            . Str::of($abstract)->replaceFirst('Illuminate\\Contracts', '')->replace('\\', ' ')->slug(),
+            $concrete,
+            $abstract
+        ))
+        ->each(fn($concrete, $abstract) => $this->app->singleton($abstract, $concrete));
```

Comment:
I feel like something like this would be more readable.

```suggestion
        $this->app->singleton(
            \Illuminate\Contracts\Http\Kernel::class,
            apply_filters('acorn/container/http-kernel', $httpKernel)
        );
        
        $this->app->singleton(
            \Illuminate\Contracts\Console\Kernel::class,
            apply_filters('acorn/container/console-kernel', \Roots\Acorn\Console\Kernel::class)
        );
        
        $this->app->singleton(
            \Illuminate\Contracts\Debug\ExceptionHandler::class,
            apply_filters('acorn/container/exception-handler', \Roots\Acorn\Console\Kernel::class)
        );
```